### ISSUE TITLE
Reference Implementation

### ISSUE THREAD

**Issue by treelzebub** [asarazan/bismarck#15]
We need a relatively realistic reference Implementation, not only for our eventual consumers but for our own ideation.

**Comment by treelzebub** [comment]
I'd like to take a stab at it this weekend and will consider the PR a request for comments and corrections :)

**Comment by asarazan** [comment]
Can you elaborate on what you mean? Do you just mean a test-bed application that is using all these features to demonstrate how they work?

**Comment by treelzebub** [comment]
Yes, a set of examples, samples, whatever you want to call them. À la https://github.com/cashapp/sqldelight/tree/master/sample

**Comment by treelzebub** [comment]
This would also help someone other than asarazan to add kdoc to the lib, which also needs to happen at some point.

**Comment by asarazan** [comment]
@vaudevillen did a little along those lines out of pure necessity to debug the ios layer. You can find it here: https://github.com/asarazan/bismarck/pull/12
